mypy/type_visitor.py

```diff
@@ -23,13 +23,13 @@
     RawExpressionType, Instance, NoneType, TypeType,
     UnionType, TypeVarType, PartialType, DeletedType, UninhabitedType, TypeVarLikeType,
     UnboundType, ErasedType, StarType, EllipsisType, TypeList, CallableArgument,
-    PlaceholderType, TypeAliasType, ParamSpecType, get_proper_type
+    PlaceholderType, TypeAliasType, ParamSpecType, SelfType, get_proper_type
 )
 
 
 @trait
 @mypyc_attr(allow_interpreted_subclasses=True)
-class TypeVisitor(Generic[T]):
+class TypeVisitor(Generic[T], metaclass=ABCMeta):
```

Comment:
Why do we need an extra metaclass here? 🤔

Reply:
I just had it in there to catch instantiations of subclasses where I hadn't implemented visit_self_type